lightning/src/ln/channelmanager.rs

```diff
@@ -1120,6 +1120,15 @@ where
 	/// `PersistenceNotifierGuard::notify_on_drop(..)` and pass the lock to it, to ensure the
 	/// Notifier the lock contains sends out a notification when the lock is released.
 	total_consistency_lock: RwLock<()>,
+	/// Tracks the progress of channels going through batch v1 channel establishment by whether
+	/// funding_signed was received and the monitor has been persisted.
+	///
+	/// This information does not need to be persisted as funding nodes can forget
```

Comment:
Discussed offline, the question here is whether we want to persist `Channel`s or information about the batch early to provide a best-effort approach to sending events such as `DiscardFunding` and being able to force-close without broadcasting. This best-effort approach should not be considered a guarantee by users though - the `ChannelManager` may or may not have been persisted asynchronously. Since it's not a guarantee, I would propose having the behavior as-is currently, and handling those channel closures on read roughly with force-closures with broadcasting here: https://github.com/lightningdevkit/rust-lightning/blob/a24626b919bf5ec5206ab65746fcee274cd589ec/lightning/src/ln/channelmanager.rs#L8535-L8545

Note that we haven't reached best-effort around emitting `DiscardFunding` in general yet. `Channel`s and `ChannelMonitor`s are only persisted at `funding_signed`, and the asynchronous persistence of `ChannelManager` and `Channel`s is not guaranteed at any point. Added a comment on the event and filed https://github.com/lightningdevkit/rust-lightning/issues/2508.